src/charm.py

```diff
@@ -93,6 +94,11 @@ def __init__(self, *args):
             self.framework.observe(self.on[rel].relation_changed, self._on_event)
 
         self._logging = LogForwarder(charm=self)
+        self._scraping = MetricsEndpointProvider(
+            self,
+            jobs=[{"static_configs": [{"targets": ["*:8080", "*:8081"]}]}],
```

Comment:
won't this try to scrape both kfam and profiles pod at 8080 and 8081?

Reply:
They are at the same pod as charm, so no. The "*" is replaced with charm Pod IP, and this pod containe 3 containers: charm, kubeflow-kfam and kubeflow-profiles.